src/components/ManageGallery/OrganizeCollection/Editor/CollectionEditor.tsx

```diff
@@ -132,7 +133,7 @@ function CollectionEditor({ queryRef, onValidChange, onHasUnsavedChange }: Props
       return;
     }
 
-    lastStagedCollection.current = stagedCollectionState;
+    lastStagedCollection.current = cloneDeep(stagedCollectionState);
```

Comment:
does this ensure that Adding Item A => Removing Item A is considered to represent "no changes"?